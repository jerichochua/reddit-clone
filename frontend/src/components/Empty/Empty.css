.empty-content {
  padding: 20px 0;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 100%;
}

.empty-message {
  margin-top: 20px;
  font-size: 14px;
  color: #999;
}
