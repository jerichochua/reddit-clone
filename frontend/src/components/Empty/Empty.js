import React from 'react';
import './Empty.css';

const Empty = ({ message }) => {
  return (
    <div className='empty-content'>
      <p>No content found. Please try a different page.</p>
      {message && <p className='empty-message'>{message}</p>}
    </div>
  );
};

export default Empty;
