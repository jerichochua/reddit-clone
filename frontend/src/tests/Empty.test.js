import renderer from 'react-test-renderer';
import Empty from '../components/Empty/Empty';

it('renders correctly with message', () => {
  const component = renderer.create(<Empty message='test message' />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
