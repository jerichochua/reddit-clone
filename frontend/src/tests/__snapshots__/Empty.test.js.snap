// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with message 1`] = `
<div
  className="empty-content"
>
  <p>
    No content found. Please try a different page.
  </p>
  <p
    className="empty-message"
  >
    test message
  </p>
</div>
`;
